packages/react-server/src/ReactFizzServer.js

```diff
@@ -609,22 +689,9 @@ function retryTask(request: Request, task: Task): void {
     return;
   }
   try {
-    let node = task.node;
-    while (
-      typeof node === 'object' &&
-      node !== null &&
-      (node: any).$$typeof === REACT_ELEMENT_TYPE &&
-      typeof node.type === 'function'
-    ) {
-      // Doing this here lets us reuse this same Segment if the next component
-      // also suspends.
-      const element: React$Element<any> = (node: any);
-      task.node = node;
-      // TODO: Classes and legacy context etc.
-      node = element.type(element.props);
-    }
```

Comment:
This is the duplication that I was aiming to get rid of. Normally I'd probably be ok duplicating this code but it's quite important that these don't go out of sync as we add new types.